### PR TITLE
Azure - Add storage utilities to support queues

### PR DESCRIPTION
**Pull request by stefangordon**
New storage utilities and tests that support upcoming notifications and blob output work.

Note I renamed `storageutils.py` to `storage_utils.py` to be correct and the PR isn't showing it as a move, basically the last two functions did not change, the rest is new.

I also added `pylru` as a py2/3 compatible attribute based LRU as I have use for it in several places.  At the moment it caches the "Get X by URI" methods so the expensive lookups only happen once per URI.

There is a queue unit test showing round trip writing/reading/deleting queue item.

### CODE REVIEW

File: tools/c7n_azure/setup.py [hunk 1]
```diff
@@ -31,5 +31,5 @@
         "custodian.resources": [
             'azure = c7n_azure.entry:initialize_azure']
     },
-    install_requires=["azure", "c7n", "click", "azure-cli-core", "adal~=0.5.0", "futures==3.1.1"],
+    install_requires=["azure", "c7n", "click", "azure-cli-core", "adal~=0.5.0", "pylru", "futures==3.1.1"],
```
Comment (kapilt) [inline tools/c7n_azure/setup.py]: so we're generally trying to keep custodian under apache license, and this pulls in a gpl dependency, which as a library in python is problematic for the aggregate work. there's a few other lru type implementations out there if your looking for something. python 3 (only) has one builtin to functools https://docs.python.org/3/library/functools.html#functools.lru_cache and there's a few dozen other libraries and recipes for them.

File: tools/c7n_azure/c7n_azure/storage_utils.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    from functools import lru_cache
+except ImportError:
+    from backports.functools_lru_cache import lru_cache
+
+
+class StorageUtilities(object):
+
+    @staticmethod
+    @lru_cache()
+    def get_storage_client_by_uri(storage_uri):
```
Comment (anastasiia-zolochevska) [inline tools/c7n_azure/c7n_azure/storage_utils.py]: I’d call it get_blob_client_by_uri as queue and table are also storage

File: tools/c7n_azure/c7n_azure/storage_utils.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        blob_service = BlockBlobService(account_name=storage_name, account_key=key)
+        blob_service.create_container(container_name)
+
+        return blob_service, container_name
+
+    @staticmethod
+    @lru_cache()
+    def get_queue_client_by_uri(queue_uri):
+        parts = urlparse(queue_uri)
```
Comment (anastasiia-zolochevska) [inline tools/c7n_azure/c7n_azure/storage_utils.py]: There is a lot of duplicate code for these 2 methods, we also might need the same methods for tables and files clients later, maybe consider reusing?

Comment (stefangordon) [reply tools/c7n_azure/c7n_azure/storage_utils.py]: Good point, I still haven't recovered from lack of sleep at PyCon :)   Will make that nicer.

**Pull request merged**